packages/elements/src/components/ino-nav-drawer/ino-nav-drawer.tsx

```diff
@@ -73,23 +84,34 @@ export class NavDrawer implements ComponentInterface {
     toggleBtn: 'Toggle Navigation',
   };
 
-  @Watch('variant')
-  variantChanged(newVariant: NavDrawerVariant) {
-    if (newVariant === 'mobile') {
-      this.activateMobileMode();
-    } else {
-      this.deactivateMobileMode();
+  /**
+   * Emits when the user clicks on the drawer toggle icon to change the open state. Contains the status in `event.detail`.
+   */
+  @Event() openChange!: EventEmitter<boolean>;
+
+  // This listener ensures that only the most recently clicked/selected list item appears as "activated"
+  @Listen('clickEl')
+  handleListItemClick(event: CustomEvent) {
+    const listItem: HTMLInoListItemElement = event.detail;
+
+    if (!listItem || listItem.tagName !== 'INO-LIST-ITEM') {
+      return;
     }
+
+    this.deactivateAllItems();
+    listItem.activated = true;
   }
 
-  componentDidLoad() {
-    this.drawerInstance = new MDCDrawer(
-      this.el.shadowRoot.querySelector('.mdc-drawer'),
-    );
-    // set initial value of open state
-    if (this.drawerInstance) {
-      this.drawerInstance.open = this.open || false;
+  componentDidUpdate() {
+    if (this.reinitialize) {
+      this.drawerInstance?.destroy();
+      this.initDrawerInstance();
+      this.reinitialize = false;
     }
+  }
```

Comment:
Why do we need the `reinitialize` flag? Can't we just reinitialize whenever the `variant` changes (so in `variantChanged`)?

Reply:
We need this, because the drawerInstance needs to initialize after the ui updates. The watcher is executed before rerendering. 

I introduced this flag, because it is not necessary to reinit the drawerInstance by every component update.

Reply:
Ah, I see. Can we name it a bit more explicit? Like `reinitializeAfterVariantChange`? Also please add a code comment on why we need this flag so we dont get confused in the future.